walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -85,18 +119,36 @@
     }
     
     // Add last two possible available time blocks if they are large enough. 
-    TimeRange availableBlock = TimeRange.fromStartEnd(availableStart, unavailableStart, false);
-    if(availableBlock.duration() >= duration) {
+    addIfLargeEnough(availableTimes, availableStart, unavailableStart, meetingDuration, false);
+    availableStart = unavailableEnd;
+    addIfLargeEnough(availableTimes, availableStart, TimeRange.END_OF_DAY, meetingDuration, true);
+    
+    return availableTimes;
+  }
+
+  // Adding a time block to a specfied collection if it is long enough for the meeting.
+  private static void addIfLargeEnough(Collection<TimeRange> availableTimes, int start, int end, 
+    int meetingDuration, boolean inclusive){
+    TimeRange availableBlock = TimeRange.fromStartEnd(start, end, inclusive);
+    if(availableBlock.duration() >= meetingDuration) {
       availableTimes.add(availableBlock);
     }
-    availableStart = unavailableEnd;
+  }
 
-    TimeRange lastBlock = TimeRange.fromStartEnd(availableStart, TimeRange.END_OF_DAY, true);
-    if(lastBlock.duration() >= duration) {
-      availableTimes.add(lastBlock);
+  // Filtering out events that don't contain any confliciting attendees.
```

Comment:
I feel like the comment on this function is not really accurate. You use the function output to filter out events with conflicts, but the function itself is just getting the conflicts.